src/database/migrations/0006-create-proposals.sql

```diff
@@ -0,0 +1,21 @@
+CREATE TABLE proposals (
+  id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  external_id VARCHAR NOT NULL,
+  applicant_id INTEGER NOT NULL,
+  opportunity_id INTEGER NOT NULL,
+  created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
+  UNIQUE(applicant_id, opportunity_id, external_id),
```

Comment:
I'm trying to guess as the best order here and it depends on what the caller will do. But in general I think the "largest granularity to finest granularity" order would also make sense, which I think is represented here with `applicant_id` followed by `opportunity_id` followed by `external_id`.

Reply:
Yeah I think that's what I had in my mind (granularity gradient).  Happy to shift it around if there's a better approach.

Reply:
Oh; I don't know if this matters at all but I just noticed that we had been putting UNIQUEs after the CONSTRAINTs.

I suspect it doesn't but I'll teak for consistency since why not.